docs/x509.rst

```diff
@@ -276,6 +292,24 @@ X.509 Certificate Object
 X.509 Extensions
 ~~~~~~~~~~~~~~~~
 
+.. class:: Extensions
+
+    .. versionadded:: 0.9
+
+    An X.509 Extensions instance is an ordered list of extensions.  The object
+    is iterable to get every extension.
+
+    .. method:: get_extension_for_oid(oid)
+
+        :param oid: An :class:`ObjectIdentifier` instance.
+
+        :returns: An instance of the extension class or None.
```

Comment:
I think exception makes sense -- empty list isn't quite the same as None
IMO. Empty list is of the same type as list with contents, None is a
different type from the extension.

On Thu, Mar 26, 2015 at 8:14 PM, Paul Kehrer <notifications@github.com>
wrote:

> In docs/x509.rst
> <https://github.com/pyca/cryptography/pull/1769#discussion_r27268346>:
>
> > @@ -276,6 +292,24 @@ X.509 Certificate Object
> >  X.509 Extensions
> >  ~~~~~~~~~~~~~~~~
> >
> > +.. class:: Extensions
> > +
> > +    .. versionadded:: 0.9
> > +
> > +    An X.509 Extensions instance is an ordered list of extensions.  The object
> > +    is iterable to get every extension.
> > +
> > +    .. method:: get_extension_for_oid(oid)
> > +
> > +        :param oid: An :class:`ObjectIdentifier` instance.
> > +
> > +        :returns: An instance of the extension class or None.
>
> Right now we return None, but I've talked a bit in the past about
> switching to an exception. It makes sense to me that if you explicitly
> request an extension and it does not exist maybe we should raise an
> exception (x509.ExtensionNotFound or something). It doesn't quite match
> the way we do get_attributes_for_oid, but that doesn't ever return none
> (only empty list).
>
> Do you have a preference?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/1769/files#r27268346>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084